cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +356,38 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubeadmVersion    string
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+
+	// Skip this check for "super-custom builds", where apimachinery/the overall codebase version is not set.
+	if kubever.KubeadmVersion == "v0.0.0" {
+		return nil, nil
+	}
+
+	kadmVersion, err := semver.ParseTolerant(kubever.KubeadmVersion)
+	if err != nil {
+		return nil, []error{fmt.Errorf("couldn't parse kubeadm version %q: %v", kubever.KubeadmVersion, err)}
+	}
+
+	k8sVersion, err := semver.ParseTolerant(kubever.KubernetesVersion)
+	if err != nil {
+		return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+	}
+
+	// Checks if k8sVersion greater or equal than the first unsupported versions by current version of kubeadm,
+	// that is major.minor+1 (all patch and pre-releases versions included)
+	firstUnsupportedVersion := semver.MustParse(fmt.Sprintf("%d.%d.%s", kadmVersion.Major, kadmVersion.Minor+1, firstPatchAndPrelease))
```

Comment:
small note; I have very similar functionality in my upgrade PR, we should unify that later in v1.9